### PR TITLE
editable notices

### LINKED ISSUE #28: Replace News with content fields that are editable by admins
Examples:

- Code of conduct
- Description
- Privacy policy

### PR DESCRIPTION
**Pull request by cryptix**
This fixes #28 

* [x] introduces _notices_ (with title, content and a language)
* [x] introuces _pinned notices_ with a fixed set of enumerated names
* [x] render notice content as markdown (using [blackfriday/v2](https://github.com/russross/blackfriday))
* [x] have some defaults (currently done by the migrations, might want to revisit with a nicer way (large text inside raw `INSERT INTO` can be finicky))
* [x] simple admin UI to edit/update notices
* [x] link them up in the UI
* [x] remove /news/

Future TODO:
1. style all the new layouts :see_no_evil:
2. styling of the markdown
3. wire up the language picking with dropdowns and session cookies
4. more tests for the HTML UI

One thing I'm side stepping for somebody else is that while you can create new notices and don't assign them to a pinned one to create custom pages, there is no way to link them in the layout but from other notices. i.e. you can't edit the menu(s) like you might be able to in a proper cms. I think this is fine _for now_.

### CODE REVIEW

File: web/handlers/http.go [hunk 1]
```diff
@@ -47,11 +57,7 @@ func New(
 		render.SetLogger(logger),
 		render.BaseTemplates("base.tmpl", "menu.tmpl"),
 		render.AddTemplates(concatTemplates(
-			[]string{
-				"landing/index.tmpl",
-				"landing/about.tmpl",
-				"error.tmpl",
-			},
+			HTMLTempaltes,
```
Comment (staltz) [inline web/handlers/http.go:60]: typo :)

**Comment by cryptix** [comment]
@staltz  could you take a look at 1 and 2 of the _future TODO_s?

There are three new templates, notice/show, notice/list and admin/notice-edit. I re-used the edit template for the _draft new translation_ mode, too.

The list and show template have links for edit when an admin is logged in. What do you think about this? I didn't want to overload the admin interface with more new pages.

I also only placed a _list of the notices_ on top in the "login nav" since we don't have a footer yet. Properly doing this would need a little template helper ala `{{notice "CodeOfConduct"}}` which looks up the notice and calls urlTo with the ID but I think this is fine for now.

I thought 4 and maybe 3 might be nice for Alex next week to get a practical intro of how everything works.

**Comment by staltz** [comment]
I'll review now :)

> could you take a look at 1 and 2 of the _future TODOs_?

Yes. Do you mean that I would do it in this PR, or in other PRs post-merging? I prefer the latter

> The list and show template have links for edit when an admin is logged in. What do you think about this? I didn't want to overload the admin interface with more new pages.

Ok. I'll take a look how the overall UI structure is, will still make up my mind how it should be presented to the user.

> I also only placed a list of the notices on top in the "login nav" since we don't have a footer yet. Properly doing this would need a little template helper ala {{notice "CodeOfConduct"}} which looks up the notice and calls urlTo with the ID but I think this is fine for now.

Yup, I thought these would be best in a footer. I'll take a look, and we can make more PRs for this stuff.

> I thought 4 and maybe 3 might be nice for Alex next week to get a practical intro of how everything works.

Good idea

**Review by staltz — APPROVED** [review]
@cryptix I think this is in a shape that it could be merged. Things I would do are:

1. Put the "list of notices" as a side-menu item called just "Notices", available only for the admin
1. Put links to CodeOfConduct and PrivacyPolicy in the footer for every user to see, using the `{{notice "CodeOfConduct"}}` idea
1. The notice page (e.g. Code of Conduct) would automatically pick up the user's declared browser language
1. The notice page would show links to alternative translations if they exist
1. Render the markdown for the `description` notice inside the landing page content

Do you think this PR enables us to do the 5th thing above? If yes, we can merge this.

**Comment by staltz** [comment]
Oh, and `/notice/list` still displays a "News" entry. Is this a side effect of me not deleting my local database and rebuilding from scratch, or is this intended?

**Comment by cryptix** [comment]
Doing these things in a new PR Post-merge is fine by me.

The News item is a new thing.. I'm not good with names and open to suggestions. The constants are defined in admindb/types.go

The markdown rendering is pretty simple, single line. Just need to pass the pindb to the handler that does the landing page and get() the notice.

**Comment by cryptix** [comment]
For 3) you can see how this is done in web/i18n - once we do this proper we need to consolidate this with the user session maybe but copying that and just using accept-language should be fine for now.

tangential: We also need a list of languages and flags somewhere

**Comment by staltz** [comment]
Great, let's merge then.

I could work on:

> - style all the new layouts see_no_evil
> - styling of the markdown
> - Put the "list of notices" as a side-menu item called just "Notices", available only for the admin
> - Put links to CodeOfConduct and PrivacyPolicy in the footer for every user to see, using the `{{notice "CodeOfConduct"}}` idea
> - Render the markdown for the description notice inside the landing page content

And these could be put in a new issue:

> - The notice page (e.g. Code of Conduct) would automatically pick up the user's declared browser language
> - The notice page would show links to alternative translations if they exist
> For 3) you can see how this is done in web/i18n - once we do this proper we need to consolidate this with the user session maybe but copying that and just using accept-language should be fine for now.
> tangential: We also need a list of languages and flags somewhere

**Pull request merged**